src/System.Net.NameResolution/tests/FunctionalTests/GetHostEntryTest.cs

```diff
@@ -86,15 +86,27 @@ public static IEnumerable<object[]> GetNoneAddresses()
             yield return new object[] { IPAddress.None };
         }
 
-        [ActiveIssue(10345, TestPlatforms.AnyUnix)]
+        [PlatformSpecific(~TestPlatforms.OSX)] // macOS will resolve IPAddress.None to broadcasthost and produce a valid listing
         [Theory]
         [MemberData(nameof(GetNoneAddresses))]
         public async Task Dns_GetHostEntryAsync_NoneIPAddress_Fail(IPAddress address)
         {
             string addressString = address.ToString();
 
-            await Assert.ThrowsAsync<SocketException>(() => Dns.GetHostEntryAsync(address));
-            await Assert.ThrowsAsync<SocketException>(() => Dns.GetHostEntryAsync(addressString));
+            await Assert.ThrowsAnyAsync<SocketException>(() => Dns.GetHostEntryAsync(address));
+            await Assert.ThrowsAnyAsync<SocketException>(() => Dns.GetHostEntryAsync(addressString));
+        }
+
+        [PlatformSpecific(TestPlatforms.OSX)] // macOS will resolve IPAddress.None to broadcasthost and produce a valid listing
+        [Theory]
+        [MemberData(nameof(GetNoneAddresses))]
+        public async Task Dns_GetHostEntryAsync_NoneIPAddress_Success(IPAddress address)
+        {
+            IPHostEntry result = await Dns.GetHostEntryAsync(address);
+            Assert.NotNull(result);
+            Assert.NotNull(result.AddressList);
+            Assert.Equal(1, result.AddressList.Length);
+            Assert.Equal(address, result.AddressList[0]);
```

Comment:
It would be odd if it weren't.

That said, we don't seem to check that "localhost" is returned when we resolve that, either.  So I don't think it matters.